sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql

```diff
@@ -32,17 +32,78 @@ stripe_customers AS (
   FROM
     mozdata.stripe.customers
 ),
-stripe_customer_country AS (
+fxa_logs_provider_country AS (
   SELECT
-    customer AS customer_id,
+    fxa_uid,
+    plan_id,
+    ARRAY_AGG(
+      STRUCT(payment_provider AS provider, source_country AS country)
+      ORDER BY
+        event_timestamp DESC
+      LIMIT
+        1
+    )[OFFSET(1)].*,
+  FROM
+    `moz-fx-data-shared-prod`.stripe_external.fxa_pay_setup_complete_v1
+  GROUP BY
+    fxa_uid,
+    plan_id
+),
+stripe_invoice_lines AS (
+  SELECT
+    lines.subscription AS subscription_id,
+    INITCAP(
+      COALESCE(
+        fxa_logs_provider_country.provider,
+        IF(
+          "paypalTransactionId" IN (SELECT key FROM UNNEST(invoices_v1.metadata)),
+          "Paypal",
+          "Stripe"
+        )
+      )
+    ) AS provider,
     LOWER(
-      -- LAST_VALUE(country ORDER BY created)
-      ARRAY_AGG(payment_method_details.card.country ORDER BY created DESC LIMIT 1)[OFFSET(0)]
+      COALESCE(fxa_logs_provider_country.country, charges_v1.payment_method_details.card.country)
     ) AS country,
+    invoices_v1.event_timestamp,
   FROM
+    `moz-fx-data-shared-prod`.stripe_external.invoices_v1
+  CROSS JOIN
+    UNNEST(lines) AS lines
+  LEFT JOIN
+    stripe_customers
+  ON
+    invoices_v1.customer = stripe_customers.customer_id
+  LEFT JOIN
     `moz-fx-data-shared-prod`.stripe_external.charges_v1
+  ON
+    invoices_v1.charge = charges_v1.id
+  LEFT JOIN
+    fxa_logs_provider_country
+  ON
+    stripe_customers.fxa_uid = fxa_logs_provider_country.fxa_uid
+    AND lines.plan.id = fxa_logs_provider_country.plan_id
+  WHERE
+    -- ignore invoices where no payment occurred
```

Comment:
When would something like this happen?

Reply:
some invoices are cancelled before they are paid (e.g. customer cancels sub after payment failure), and some invoices have a negative outstanding balance (from a refund), which can result in an invoice where `amount_due = 0`.